### PR TITLE
MM-28403 Cypress/E2E: Update all tests references from flag to save

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Update all tests references from flag to save.

Note: I didn't update application code related to flag 'cause it's a big task. Will update in the future once team owner (webapp/platform) updates those.

#### Ticket Link
Jira ticket: https://mattermost.atlassian.net/browse/MM-28403

### CODE REVIEW

**Review by josephbaylon — APPROVED** [review]
LGTM

**Review by jgilliam17 — APPROVED** [review]
Thank you @saturninoabril 
Looks good to merge.
- QA note: Test case details and links updated.

File: e2e/cypress/integration/messaging/post_header_spec.js [hunk 1]
```diff
@@ -164,7 +164,7 @@ describe('Post Header', () => {
             cy.get('#searchResultsCloseButton').should('be.visible').click();
         });
 
-        // # Click on the "Flagged Posts" icon to the right of the "Search" box
+        // # Click on the "Saved Posts" icon to the right of the "Search" box
         cy.get('#channelHeaderFlagButton').click();
 
         // * Verify that the RHS for pinned posts is opened.
```
Comment (jgilliam17) [inline e2e/cypress/integration/messaging/post_header_spec.js:170]:
```suggestion
        // * Verify that the RHS for saved posts is opened.
```

**Review by jgilliam17 — COMMENTED** [review]
Sorry, missed this comment update earlier 
There's also a `pinned` ~in~ on line 171, that I think should be `saved`, for some reason it is not allowing me to add a suggestion there.

**Pull request merged**